table/block_based/block_based_table_reader.cc

```diff
@@ -152,15 +152,16 @@ Status ReadAndParseBlockFromFile(
     BlockCreateContext& create_context, bool maybe_compressed,
     const UncompressionDict& uncompression_dict,
     const PersistentCacheOptions& cache_options,
-    MemoryAllocator* memory_allocator, bool for_compaction, bool async_read) {
+    MemoryAllocator* memory_allocator, bool for_compaction,
+    const std::shared_ptr<Compressor>& compressor, bool async_read) {
```

Comment:
I consider `const shared_ptr<X>&` parameters to be an anti-pattern. It prevents move semantics where it might be useful, and is non-committal about intention to save a reference. I suggest deciding between `const X&` and plain `shared_ptr<X>`.

Reply:
I see the "mixed message" of passing const shared_ptr<T>&. I'll review all occurrences and verify if they're justified (e.g., if a copy of the shared_ptr is eventually made).